testutil/timing.go

```diff
@@ -0,0 +1,21 @@
+package testutil
+
+import (
+	"flag"
+	"testing"
+)
+
+// We can't run timing-sensitive tests in CI because of the
+// great variance in runner performance. Instead of not testing timing at all,
+// we relegate it to humans manually running certain tests with the "-timing"
+// flag from time to time.
+//
+// Eventually, we should run all timing tests in a self-hosted runner.
+
+var timingFlag = flag.Bool("timing", false, "run timing-sensitive tests")
```

Comment:
I think we won't be able to do `go test ./... -timing` unless all test packages import testutil, a shortcoming of using flags in tests. I think the only way around it would be to use build tags but maybe this won't be an issue.

Reply:
Hmm, I explored build tags but decided against it because it could lead to file bloat.

Let me know what you think about [this approach](https://github.com/coder/coder/pull/7237/commits/8de8ffd574736cc0798db1d32fb8ff32bab4025a).

Reply:
> Hmm, I explored build tags but decided against it because it could lead to file bloat.

I was thinking it could be contained to the `testutil` package using the checking function you already implemented, so only 1 more file should be necessary. The build tag would then decide if the test becomes enabled or not.

Example:

```
//go:build timing

package testutil

var _ = func() any {
	timing = true
	return nil
}()
```

(Or use `init`, but the above avoids using that reserved name.)

This would allow running the entire test suite with `go test ./... -tags=timing`.

> Let me know what you think about [ba6f151](https://github.com/coder/coder/pull/7237/commits/ba6f151a5882da3144d6b2f2df5d8f455a3ef033).

I think that looks alright since it requires an intent to run, otherwise generating go_tests.json would be too slow.

Reply:
Nice idea, let's go that way. See new code.